pkg/operator/hive/hive.go

```diff
@@ -540,46 +540,46 @@ func monitoringLabelPatch(enable bool) string {
 	}
 }
 
-// deployMonitoring enables metrics reporting.
-// NOTE: When changing this method, be sure to keep undeployMonitoring in sync.
-func (r *ReconcileHiveConfig) deployMonitoring(hLog log.FieldLogger, h resource.Helper, instance *hivev1.HiveConfig) error {
+// reconcileMonitoring switches metrics exporting on or off, according to HiveConfig.Spec.ExportMetrics
+func (r *ReconcileHiveConfig) reconcileMonitoring(hLog log.FieldLogger, h resource.Helper, instance *hivev1.HiveConfig) error {
 	hiveNSName := getHiveNamespace(instance)
+	enable := instance.Spec.ExportMetrics
 
-	// setup the hiveNSName labels so that openshift-prometheus discovers the namespace.
-	if err := h.Patch(types.NamespacedName{Name: hiveNSName}, "Namespace", "v1", []byte(monitoringLabelPatch(true)), ""); err != nil {
-		hLog.WithError(err).Error("error updating hive namespace label to enable monitoring")
-	}
-
-	// Load namespaced assets, decode them, set to our target namespace, and apply:
-	for _, assetPath := range monitoringNamespacedAssets {
-		if err := util.ApplyAssetWithNSOverrideAndGC(h, assetPath, hiveNSName, instance); err != nil {
-			hLog.WithError(err).Error("error applying object with namespace override")
-			return err
-		}
-		hLog.WithField("asset", assetPath).Info("applied asset with namespace override")
+	var labelVal, verb1, verb2, verb3 string
```

Comment:
can we call those verbs something like 'desiredStateVerb' 'progressingVerb' 'completedVerb' , or just anything to make their purpose a bit more clear?